apps/tenant-management-webapp/src/app/pages/admin/services/form/definitions/addEditFormDefinitionEditor.tsx

```diff
@@ -847,33 +855,26 @@ export function AddEditFormDefinitionEditor(): JSX.Element {
         }}
       />
       <AddEditDispositionModal
-        open={selectedEditModalIndex !== null}
-        isEdit={true}
+        open={selectedEditModalIndex !== null || newDisposition}
+        isEdit={selectedEditModalIndex !== null}
         existingDispositions={definition?.dispositionStates}
-        initialValue={definition?.dispositionStates && definition.dispositionStates[selectedEditModalIndex]}
+        initialValue={getDispositionForModal()}
         onSave={(currentDispositions) => {
-          definition.dispositionStates[selectedEditModalIndex] = currentDispositions;
-
-          setDefinition(definition);
-        }}
-        onClose={() => {
-          setSelectedEditModalIndex(null);
-        }}
-      />
-      <AddEditDispositionModal
-        open={newDisposition}
-        isEdit={false}
-        existingDispositions={definition?.dispositionStates}
-        initialValue={definition?.dispositionStates && definition.dispositionStates[selectedEditModalIndex]}
-        onSave={(currentDispositions) => {
-          const dispositionStates = definition.dispositionStates || [];
-          const tempDefinition = { ...definition };
-          dispositionStates.push(currentDispositions);
-          tempDefinition.dispositionStates = dispositionStates;
-          setDefinition(tempDefinition);
+          if (newDisposition) {
+            const dispositionStates = definition.dispositionStates || [];
+            const tempDefinition = { ...definition };
+            dispositionStates.push(currentDispositions);
+            tempDefinition.dispositionStates = dispositionStates;
+            setDefinition(tempDefinition);
+          } else {
```

Comment:
To make this code cleaner you should refactor and create a function, i.e.
const saveDisposition = (currentDispositions, etc...) => { onSave code goes here}
and then onSave=saveDisposition
As it is, you have no way to unit test this code.

Reply:
Hi @rstyan, can you take another look at it.